### ISSUE TITLE
Compile to OS X 

### ISSUE THREAD

**Issue by mathieutozer** [martijnwalraven/meteor-ios#4]
After a very quick glance at the code, it seems as though this could be made to target OS X. What do you think?

**Comment by martijnwalraven** [comment]
I haven't tested it myself, but I don't see any principled reason why it couldn't work on OS X. Out of the top of my head the only real dependency right now is the application lifecycle and network reachability code. It shouldn't be too difficult to factor that out. I've actually stopped using CocoaPods in my own code and will add a project with a framework target soon. That should make it easier to support a separate OS X target.

**Comment by mathieutozer** [comment]
I took a closer look and yes I think with some compile time flags it could
be easy to do.

**Comment by gazzer82** [comment]
Do you think this is likely to happen? I am about to start a project that will have an iOS and OS X application and would very much like to use this framework in them both?

Thanks

Gareth

**Comment by martijnwalraven** [comment]
You'd have to spend some time removing iOS dependencies from the code and configuring the CocoaPods spec for both platforms (or set up separate framework targets), but I don't foresee any real trouble. I'm not sure when I'll get to this myself, as I'm not working on OS X apps right now, but it is on my to do list.

**Comment by mathieutozer** [comment]
I took a look at it myself and found that the SimpleKeychain dependency to be the non trivial part. Everything else was pretty simple. Martjin what role does the keychain play? Could it be sidestepped for now so that we could move forward?

**Comment by mathieutozer** [comment]
I've got it to compile by compiling out iOS specific pieces, and without having to comment out too much code ;)

I'm going to take it for a spin in one of my projects and if all goes well I'll put up a pull request.

**Comment by martijnwalraven** [comment]
Let me know how it goes. I'm in the process of making a few other changes to the project structure, so maybe you should wait for them to land before working on a pull request. We would also have to add separate targets for iOS and OS X, and change the Podspec.

**Comment by mathieutozer** [comment]
Yep, I've already updated the podspec. The changes were minimal (they took
maybe 10 minutes to compile out with compile time flags) so go ahead.

**Comment by Dev1an** [comment]
Mathieu, can you publish and example OS X project that uses this meteor framework?

**Comment by mathieutozer** [comment]
I had to update a sub dependency to support the OS X version of meteor-ios, so I guess I have to have that pulled first and then make another for meteor-ios.

Embarrassingly this is my first github pull request

**Comment by mathieutozer** [comment]
First pull request for SimpleKeychain in case anyone wants to take a look 

https://github.com/mathieutozer/SimpleKeychain/commit/2b6d2604be84b2a1f52299e615921ca4eb9697e0?diff=split

**Comment by mathieutozer** [comment]
Working version of Leaderboard for OS X
![screen shot 2015-04-28 at 4 59 11 pm](https://cloud.githubusercontent.com/assets/18321/7382711/f4133ca2-edc7-11e4-8caa-cec1ad7cd061.png)

**Comment by Dev1an** [comment]
Can you put the source code of your OSX leaderbord example online?

**Comment by mathieutozer** [comment]
I was waiting for the pull request for simple keychain to get done, do you
have a local working version of meteor for os x running and want to see how
to make a mac os x leaderboard app or can you wait until the official pull
request is ready?

**Comment by mathieutozer** [comment]
Ok I've pushed a shaky version to https://github.com/mathieutozer/meteor-ios

I am still waiting for a pull request for SimpleKeychain to happen before making an official PR for meteor-ios, I can't work out how to reference my fork with the podspecs (and some googling tells me it is not possible anyway) so I just copied the keychain source into the meteor pod for now to share how this looks. 

Also for some reason with this change the test target in the generated workspace is screwed up. If anyone can clone take a look that would be appreciated!

**Comment by ssteinerx** [comment]
Any more progress on this?  I can help; I have a big paying project that could use this.

**Comment by mathieutozer** [comment]
Hey I haven't worked on this for a while, I ran into a pretty nasty rebase. If you want to pick it up while I'm on vacation, go ahead!

**Comment by ssteinerx** [comment]
What's the nasty rebase, where are you at, and what can I do to help move it along?

**Comment by mathieutozer** [comment]
Just rebasing my fork onto the latest master was pretty bad - maybe start
there?

**Comment by alminde** [comment]
+1

**Comment by mathieutozer** [comment]
For people following along the current master in my branch should work. Someone try it out?

**Comment by alminde** [comment]
Cool, @mathieutozer. I’ll check it out asap!

Does it include your OS X leaderboard sample?

**Comment by mathieutozer** [comment]
Yes it does!

**Comment by alminde** [comment]
DUDE, SWEET!
